File: C++/프로그래머스/1/131705. 삼총사/삼총사.cpp
#include <iostream>
#include <vector>

using namespace std;

int answer, N;
vector<int> number;

void combination(int start, int count, int ssum) {
    if(count == 3) {
        if(ssum == 0) answer++;
        return;
    }
    for(int i = start; i < N; i++) {
        combination(i + 1, count + 1, ssum + number[i]);
    }
}


int solution(vector<int> number) {
    ::number = number;
    N = number.size();
    
    combination(0, 0, 0);
    return answer;
}

File: Python3/프로그래머스/1/131705. 삼총사/삼총사.py
answer = 0
N = 0

def combination(start: int, count: int, ssum: int):
    if count == 3:
        if ssum == 0:
            global answer
            answer += 1
        return
    
    for i in range(start, N):
        combination(i + 1, count + 1, ssum + gnumber[i])

def solution(number):
    global N
    global gnumber
    
    N = len(number)
    gnumber = number
    
    combination(0, 0, 0)
    
    return answer